lib/baseProducer.js

```diff
@@ -134,7 +134,7 @@ BaseProducer.prototype.buildPayloads = function (payloads, topicMetadata) {
       if (message instanceof KeyedMessage) {
         return message;
       }
-      return new Message(0, 0, '', message);
+      return new Message(0, 0, p.key || '', message);
```

Comment:
Yes, even if I give a pass here, it would still having problem at `encodeMessage()`. So, I guess the `key` must be type of `String`, right? In such case, what if `p.key = true`? I think the boolean value will cause some problem, I think we should use `_.isEmpty(p.key)` to check it.